src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -34,8 +34,14 @@ public int GetCodeContext(ulong uCodeLocationId, out IDebugCodeContext2 ppCodeCo
         public int GetCodeLocationId(IDebugCodeContext2 pCodeContext, out ulong puCodeLocationId)
         {
             AD7MemoryAddress addr = pCodeContext as AD7MemoryAddress;
-            puCodeLocationId = addr.Address;
-            return Constants.S_OK;
+            if (addr != null)
+            {
+                puCodeLocationId = addr.Address;
+                return Constants.S_OK;
+            }
+
+            puCodeLocationId = 0;
+            return Constants.S_FALSE;
```

Comment:
>S_FALSE [](http://example.com/codeflow?start=29&length=7)

Looking at how the other Microsoft debug engines handle this, I would suggest using a failure code here. You can just use `Constants.E_FAIL`.